### ISSUE TITLE
Wire SMS activation code does not change.

### ISSUE THREAD

**Issue by legovini** [wireapp/wire-android#189]
If I register to wire, delete my account, and then register again, the activation code I receive via SMS is always the same. It is expected to be random for obvious security reasons.

**Comment by someoneEsle** [comment]
Just so you know, they're aware of it: https://twitter.com/gillo/status/709313322672398336

**Comment by legovini** [comment]
Thanks @someoneEsle. For the record, they say

> Hi, A new code is generated if fails 5 times, or if sent to a new phone number.

but I don't see the reason for this and seems a bad idea to me. It's easy to peek into one's past SMS messages, read the activation code and re-register with the same phone number on another device. Moreover, phone companies can (and do) log SMS messages, so an easy way to hijack a Wire account is available to whoever has access to those logs.

**Comment by someoneEsle** [comment]
True, I'm pretty sure they're aware of the implications and it's a matter of time before they fix it. By the way you get an email and a notification every time someone registers a new device, but your point still makes sense.

**Comment by ImAnnoying2** [comment]
Is this issue still relevant?

**Comment by marcoconti83** [comment]
Hi everyone,
thanks for contributing to the conversation. The SMS code is not generated by the Android application but by the wire server application. Moving this issue there.
